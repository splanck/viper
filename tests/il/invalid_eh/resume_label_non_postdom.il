il 0.1.2
func @resume_label_non_postdom(i1 %flag, Error %err, ResumeTok %tok) -> void {
entry:
  eh.push ^handler
  cbr %flag, ^left, ^right

left:
  eh.pop
  br exit

right:
  %tmp:i64 = sdiv.chk0 1, 1
  eh.pop
  br exit

exit:
  ret

handler ^handler(%err:Error, %tok:ResumeTok):
  eh.entry
  resume.label %tok, ^left
}
