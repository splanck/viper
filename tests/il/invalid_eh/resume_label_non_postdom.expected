error: verify.eh.resume_label_target: resume_label_non_postdom:handler: resume.label %t5 label left: target ^left must postdominate block right
